math/src/main/java/com/powsybl/math/graph/UndirectedGraphImpl.java

```diff
@@ -129,6 +129,24 @@ public int addVertex() {
         return v;
     }
 
+    @Override
+    public void addVertexIfNotPresent(int v) {
+        if (v < vertices.size()) {
+            if (removedVertices.contains(v)) {
```

Comment:
Are you sure this is efficient: should we store the removedVertices in another container (ordered) like a set?